src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1526,7 +1518,18 @@ internal bool IsSubmissionSyntaxTree(SyntaxTree tree)
         /// <summary>
         /// Global imports (including those from previous submissions, if there are any).
         /// </summary>
-        internal ImmutableArray<NamespaceOrTypeAndUsingDirective> GlobalImports => _globalImports.Value;
+        internal ImmutableArray<NamespaceOrTypeAndUsingDirective> GlobalImports
+        {
+            get
+            {
+                if (_globalImports.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _globalImports, BindGlobalImports());
```

Comment:
💡 `InterlockedOperations` has overloads to directly support this case

Reply:
Without the IsDefault check, won't ever call into InterlockedOperations.Initialize require a closure allocation?

Reply:
➡️ Covered by https://github.com/dotnet/roslyn/pull/72371#discussion_r1511749668